apps/wallet/src/ui/app/hooks/useGetNFTMeta.ts

```diff
@@ -13,23 +14,12 @@ export type NFTMetadata = {
 };
 
 export function useGetNFTMeta(objectID: string) {
-    const data = useGetObject(objectID);
-
+    const data = useGetObject(objectID, { showDisplay: true });
     const nftMeta = useMemo(() => {
         if (!data.data) return null;
-
         const { details } = data.data || {};
-        if (!is(details, SuiObjectData) || !data) return null;
-        const fields = getObjectFields(data.data);
-        if (!fields?.url) return null;
-        return {
-            description:
-                typeof fields.description === 'string'
-                    ? fields.description
-                    : null,
-            name: typeof fields.name === 'string' ? fields.name : null,
-            url: fields.url,
-        };
+        if (!is(details, SuiObjectData) || !details.display) return null;
+        return processDisplay(details.display);
```

Comment:
If you nuke the `processDisplay` from core, you could just move it here basically, since this is already basically a view-model approach to this problem (which is fine for getting this out, just don't want to proliferate it by moving it to core).